### PR TITLE
CGAL_imageIO: Fix .inr writing without zlib

### PR DESCRIPTION
**Pull request by maxGimeno**
## Summary of Changes
When linking with zlib, always use the gz API, as gzFiles are not compatible with fopen(), fread(),etc. 
## Release Management

* Affected package(s):CGAL_imageIO

### CODE REVIEW

**Comment by lrineau** [comment]
There is a problem. [`gzfwrite` was only added in zlib recently](https://zlib.net/ChangeLog.txt):
>  Changes in 1.2.9 (31 Dec 2016)
>  - [...]
>  - Add `gzfwrite()`, duplicating the interface of `fwrite()`

Your new code will have to use preprocessor conditionals.

File: CGAL_ImageIO/include/CGAL/ImageIO_impl.h [hunk 1]
```diff
@@ -63,6 +63,12 @@
 #include <CGAL/ImageIO/mincio.h>
 #endif
 
+
+#ifdef CGAL_USE_ZLIB
+#define ZLIB_MAJOR_VERSION TEST_VERNUM>>8
```
Comment (lrineau) [inline CGAL_ImageIO/include/CGAL/ImageIO_impl.h]: What is `TEST_VERNUM`?

Comment (maxGimeno) [reply CGAL_ImageIO/include/CGAL/ImageIO_impl.h]: A copy/paste error :(

**Comment by maxGimeno** [comment]
from zlib manual : 
#define ZLIB_VERNUM 0x12b0

So it is 1.2.11->1211->x012b0.
With bit shifting I can split it in a major-minor/super-minor versions and test like this.

**Comment by lrineau** [comment]
Successfully tested in [CGAL-4.13-Ic-97][*], and then merged.

  [*]: https://cgal.geometryfactory.com/CGAL/testsuite/results-4.13-Ic-97.shtml

**Pull request merged**